antsibull/data/collection-enum.py

```diff
@@ -124,6 +128,39 @@ def load_all_plugins(plugin_type, basedir, coll_filter):
     return result
 
 
+def load_collection_meta_manifest(b_manifest_path, data):
+    with open(b_manifest_path, 'rb') as f:
+        meta = json.load(f)
+    data['namespace'] = meta['collection_info']['namespace']
+    data['name'] = meta['collection_info']['name']
+    data['version'] = meta['collection_info']['version']
+
+
+def load_collection_meta_galaxy(b_galaxy_path, data):
+    with open(b_galaxy_path, 'rb') as f:
+        meta = yaml.safe_load(f)
+    data['namespace'] = meta['namespace']
+    data['name'] = meta['name']
+    data['version'] = meta.get('version')
+
+
+def load_collection_meta(b_path):
+    data = {
+        'version': None,
```

Comment:
Honestly, it's unlikely that the version wouldn't be specified. What case does this cover?

Reply:
Collections checked out from source where the version is inserted into `galaxy.yml` on build. That's for example the case in ansible.netcommon: https://github.com/ansible-collections/ansible.netcommon/blob/main/galaxy.yml#L13

Reply:
Interesting. Curious if this'd be important in the context of https://github.com/ansible/ansible/pull/72591...

Reply:
It's maybe a good idea to have that collection repo checked out and included in your COLLECTIONS_PATHS.

Reply:
There might be even collections repos without `galaxy.yml` and `MANIFEST.json` (I think I saw some, but cannot remember which ones). No idea if that actually works with current Ansible, but it's probably worth testing that as well.

Reply:
`ansible-galaxy` and `ansible-doc` look for metadata files, but they are completely functional without any.